src/main/java/org/jabref/gui/BasePanel.java

```diff
@@ -525,7 +525,7 @@ public void update() {
 
         actions.put(Actions.OPEN_URL, new OpenURLAction());
 
-        actions.put(Actions.MERGE_WITH_FETCHED_ENTRY, new MergeWithFetchedEntryAction(this));
+        actions.put(Actions.MERGE_WITH_FETCHED_ENTRY, new MergeWithFetchedEntryAction(this, frame));
```

Comment:
could you please pass the `DialogService` directly, instead of the frame (makes the dependency clearer).